src/libstd/sys/windows/stdio.rs

```diff
@@ -134,6 +134,28 @@ impl Stdin {
     pub fn new() -> io::Result<Stdin> {
         Ok(Stdin { surrogate: 0 })
     }
+    pub unsafe fn set_nonblocking(&self, nonblocking: bool) -> io::Result<()> {
+        let handle = get_handle(c::STD_INPUT_HANDLE)?;
+        let mut mode = 0;
+        if !c::GetConsoleMode(handle, &mut mode) {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "Cannot get console mode for the given handle",
+            ));
+        }
+        if nonblocking {
+            mode ^= c::ENABLE_LINE_INPUT;
```

Comment:
Won't doing xor cause `ENABLE_LINE_INPUT` to be set if it wasn't already set?

Reply:
Good point.

Reply:
Perhaps it might be clearer to do `mode &= !c::ENABLE_LINE_INPUT;`?

Reply:
Not sure it would work as expected though...

Reply:
Why not?

Reply:
You're absolutely right, my bad... For further reference, I tested using [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e84c215a091145526938873fa5c793a3) to confirm it works.